polygeist/lib/Dialect/Polygeist/Transforms/BareMemRefToLLVM.cpp

```diff
@@ -655,14 +655,23 @@ void mlir::polygeist::populateBareMemRefToLLVMConversionPatterns(
         if (!canBeLoweredToBarePtr(type))
           return std::nullopt;
 
+        FailureOr<unsigned> addrSpace = converter.getMemRefAddressSpace(type);
+        if (failed(addrSpace)) {
+          emitError(UnknownLoc::get(type.getContext()),
+                    "conversion of memref memory space ")
+              << type.getMemorySpace()
+              << " to integer address space failed. Consider adding memory "
+                 "space conversions.";
+          return Type{};
+        }
```

Comment:
Basically took this from the non-bare case [here](https://github.com/intel/llvm/blob/50f1605d2eb1c89ea964d0730521aa532fc7c30e/mlir/lib/Conversion/LLVMCommon/TypeConverter.cpp#L326)